### PR TITLE
MAINT, SIMD: Handle overflow gracefully in integer division

### PR DESCRIPTION
**Pull request by ganesh-k13**
### Handle SIMD division overflow
Merge before: https://github.com/numpy/numpy/pull/21507
Part of: https://github.com/numpy/numpy/issues/21506

Raising this separately to not blow up the original PR on scalars.

### Changes:
Before:
```py
>>> import numpy as np
>>> np.array([np.iinfo(np.int32).min]*10, dtype=np.int32) // np.int32(-1)
<stdin>:1: RuntimeWarning: divide by zero encountered in floor_divide
array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int32)
```

Now:
```py
>>> import numpy as np
>>> np.array([np.iinfo(np.int32).min]*10, dtype=np.int32) // np.int32(-1)
<stdin>:1: RuntimeWarning: overflow encountered in floor_divide
array([-2147483648, -2147483648, -2147483648, -2147483648, -2147483648,
       -2147483648, -2147483648, -2147483648, -2147483648, -2147483648],
      dtype=int32)
```

### CODE REVIEW

**Comment by ganesh-k13** [comment]
Hey @seberg, thought of going bottom up on this one. This solves the SIMD overflows, once merged, I'll fix the scalars and arrays for other ops along with the tests.

**Review by seberg — COMMENTED** [review]
Thanks, happy with doing this one-by-one.  I am slightly wondering if/how we should backport it, but maybe that is better seen later.

LGTM modulo the test fixup.  Will wait a bit in case someone more familiar with SIMD wants to have a look.

**Comment by charris** [comment]
At this point I don't want to do anymore backports except straight bug fixes.

**Comment by seberg** [comment]
Yeah, agreed.  It may make sense to do a very minimal bug-fix that just solves the crashed.  But maybe lets figure out this stuff here first and then do that?

File: numpy/core/src/umath/loops_arithmetic.dispatch.c.src
Comment (ganesh-k13) [reply numpy/core/src/umath/loops_arithmetic.dispatch.c.src:355]: Ah you are right, shouldn't code after 12 AM :)

Comment (ganesh-k13) [reply numpy/core/src/umath/loops_arithmetic.dispatch.c.src:355]: Ah you are right, shouldn't code after 12 AM :)

File: numpy/core/tests/test_umath.py
Comment (ganesh-k13) [reply numpy/core/tests/test_umath.py:367]: Ohh ok, thanks!

File: numpy/core/src/umath/loops_arithmetic.dispatch.c.src
Comment (ganesh-k13) [inline numpy/core/src/umath/loops_arithmetic.dispatch.c.src:273]: I hope VSX is covered as part of the CI. Else would be nice to get @seiko2plus opinion on this?

**Review by seberg — APPROVED** [review]
Thanks, LGTM now.  Will give it a bit time in case of SIMD related review.

**Comment by charris** [comment]
The test failure can be ignored.

File: numpy/core/src/umath/loops_arithmetic.dispatch.c.src
Comment (rafaelcfsousa) [inline numpy/core/src/umath/loops_arithmetic.dispatch.c.src:286]: @ganesh-k13 : there is a missing `@` here:

```suggestion
                      quo = npyv_select_@sfx@(overflow, vmin, quo);
```


Applying the change above (the missing `@`), all tests pass on VSX4/Power10

I also see the warning message on VSX4/Power10 using the example shared above.
```
<stdin>:1: RuntimeWarning: overflow encountered in floor_divide
array([-2147483648, -2147483648, -2147483648, -2147483648, -2147483648,
       -2147483648, -2147483648, -2147483648, -2147483648, -2147483648],
      dtype=int32)
```

Comment (seberg) [reply numpy/core/src/umath/loops_arithmetic.dispatch.c.src]: Wooops :), thanks for having a look!  Planing on merging when tests are done (again).

Comment (ganesh-k13) [reply numpy/core/src/umath/loops_arithmetic.dispatch.c.src]: Wow, my bad! Thanks a lot!

**Pull request merged**
